packages/ecs-browser/src/Browser.tsx

```diff
@@ -14,26 +14,29 @@ import { Coord } from "./shared";
  */
 export const Browser = observer(
   ({
-    entities,
     layers,
     setContractComponentValue,
-    devHighlightComponent,
-    hoverHighlightComponent,
     prototypeComponent,
     nameComponent,
     spawnPrototypeAt,
     world,
   }: {
-    entities: EntityID[];
     layers: Layers;
-    setContractComponentValue: SetContractComponentFunction<Schema>;
-    devHighlightComponent: Component<{ value: Type.OptionalNumber }>;
-    hoverHighlightComponent: Component<{ x: Type.OptionalNumber; y: Type.OptionalNumber }>;
+    setContractComponentValue?: SetContractComponentFunction<Schema>;
     prototypeComponent?: Component<{ value: Type.StringArray }>;
     nameComponent?: Component<{ value: Type.String }>;
     spawnPrototypeAt?: (prototypeId: EntityID, position: Coord) => void;
     world: World;
   }) => {
+    const devHighlightComponent = defineComponent(world, {
+      value: Type.OptionalNumber,
+    });
+
+    const hoverHighlightComponent = defineComponent(world, {
```

Comment:
I agree that it is cleaner to construct the dev only components inside of the browser itself, but right now we are losing the ability to react (easily) to these component changes during rendering.

In Sky Strife currently the HoverHighlight component is being set as the user's mouse is moving and then that information gets passed into here. I could get the same functionality by grabbing the created Component from this package and hooking it into the existing systems.

I see 2 options:
1. We create the Components outside of the Browser using a helper function (`createBrowserDevComponents` or something) and then pass them in. This is at least easier to adhere to the interface the Browser expects.
2. Create `getDevComponents` helper that gives the game layers access to the Components.

Right now to make it work we would have to grab the components from the World and manually type them. It feels strange because users would have to know these magic components got created and their typings.

Reply:
Ahh word!

I'll build out option 1, 

I think thats the best option, I didn't think about them being used outside of the Browser component. Good call!